File: src/AdventureWorks.Application/DataEngine/Sales/Currency/Queries/GetCurrencies/CurrencyLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.Sales;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;


namespace AdventureWorks.Application.DataEngine.Sales.Currency.Queries.GetCurrencies
{
    public partial class CurrencyLookupModel : IHaveCustomMapping
    {
        public string CurrencyCode { get; set; }
        public string Name { get; set; }
        public DateTime ModifiedDate { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.Sales.Currency, CurrencyLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<CurrencyLookupModel, BaseCurrency>().IgnoreMissingDestinationMembers();
        }
    }
}

File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/sales/currency/currency-lookup-model.ts

export interface ICurrencyLookupModel {
  currencyCode: string;
  name: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class CurrencyLookupModel implements ICurrencyLookupModel {
  currencyCode: string;
  name: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<CurrencyLookupModel>) {
    if (!!init) {
      Object.assign<CurrencyLookupModel, Partial<CurrencyLookupModel>>(this, init);
    }
  }
}
